### PR TITLE
test: add e2e tests for navigating to multiple views in the same execution stack

### PR DESCRIPTION
**Pull request by mofogasy**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](https://github.com/SchweizerischeBundesbahnen/scion-workbench/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added
- [ ] Docs have been added or updated

## PR Type
What kind of change does this PR introduce?

- [ ] Fix
- [ ] Feature
- [ ] Documentation
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Performance (changes that improve performance)
- [x] Test (adding missing tests, refactoring tests; no production code change)
- [ ] Chore (other changes like formatting, updating the license, removal of deprecations, etc)
- [ ] Deps (changes related to updating dependencies)
- [ ] CI (changes to our CI configuration files and scripts)
- [ ] Revert (revert of a previous commit)
- [ ] Release (publish a new release)
- [ ] Other... Please describe:

## Issue

Issue Number: #254 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by danielwiehl — CHANGES_REQUESTED** [review]

File: apps/workbench-client-testing-app/src/app/activator/activator.module.ts [hunk 1]
```diff
@@ -201,5 +201,23 @@ export class ActivatorModule {
...
+      type: WorkbenchCapabilities.View,
+      qualifier: {
+        component: 'bulk-navigation',
+        app,
+      },
+      description: '[e2e] Allows testing bulk navigation',
+      private: false,
+      properties: {
+        path: 'test-bulk-navigation',
+        pinToStartPage: true,
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply apps/workbench-client-testing-app/src/app/activator/activator.module.ts:216]: For now, I think we can go with the first approach. We should keep in mind the second approach though, as soon as more test pages are added.

File: apps/workbench-client-testing-app/src/app/bulk-navigation-page/bulk-navigation-page.component.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+:host {
+  display: grid;
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply apps/workbench-client-testing-app/src/app/bulk-navigation-page/bulk-navigation-page.component.scss:2]: Done

File: projects/scion/e2e-testing/src/workbench-client/bulk-navigation.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import {test} from '../fixtures';
+import {BulkNavigationPagePO} from './page-object/bulk-navigation-page.po';
+import {expect} from '@playwright/test';
+
+test.describe('Bulk Navigation', () => {
+
+  test('should navigate to multiple views if waiting for each navigation to complete', async ({appPO, microfrontendNavigator}) => {
+    await appPO.navigateTo({microfrontendSupport: true});
+
+    const bulkNavigationPagePO = await microfrontendNavigator.openInNewTab(BulkNavigationPagePO, 'app1');
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply projects/scion/e2e-testing/src/workbench-client/bulk-navigation.e2e-spec.ts:20]: Done

File: projects/scion/e2e-testing/src/workbench-client/page-object/bulk-navigation-page.po.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    await assertElementVisible(this._locator);
+    await this._locator.locator('button.e2e-navigate').click();
+  }
+
+  public async clickNavigateAwait(): Promise<void> {
+    await assertElementVisible(this._locator);
+    await this._locator.locator('button.e2e-navigate-await').click();
+    // Waiting for each navigation to complete takes some time.
+    // Therefore, the 10th view tab is awaited before continuing.
+    await this._appPO.page.locator(`wb-view-tab.e2e-test-view`).nth(9).waitFor({state: 'attached'});
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply projects/scion/e2e-testing/src/workbench-client/page-object/bulk-navigation-page.po.ts:34]: Done

File: projects/scion/e2e-testing/src/workbench-client/page-object/bulk-navigation-page.po.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import {ElementSelectors} from '../../helper/element-selectors';
+
+export class BulkNavigationPagePO {
+
+  private readonly _locator: Locator;
+
+  constructor(private _appPO: AppPO, viewId: string) {
+    this._locator = this._appPO.page.frameLocator(ElementSelectors.routerOutlet(viewId)).locator('app-bulk-navigation-page');
+  }
+
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply projects/scion/e2e-testing/src/workbench-client/page-object/bulk-navigation-page.po.ts:23]: Done

File: projects/scion/e2e-testing/src/workbench/page-object/bulk-navigation-page.po.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+export class BulkNavigationPagePO {
+
+  private readonly _locator: Locator;
+
+  constructor(private _appPO: AppPO, viewId: string) {
+    this._locator = this._appPO.findView({viewId: viewId}).locator('app-bulk-navigation-page');
+  }
+
+  public async clickNavigateNoAwait(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply projects/scion/e2e-testing/src/workbench/page-object/bulk-navigation-page.po.ts:23]: Done

File: projects/scion/e2e-testing/src/workbench/bulk-navigation.e2e-spec.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+test.describe('Bulk Navigation', () => {
+
+  test('should navigate to multiple views if waiting for each navigation to complete', async ({appPO, workbenchNavigator}) => {
+    await appPO.navigateTo({microfrontendSupport: false});
+
+    const bulkNavigationPagePO = await workbenchNavigator.openInNewTab(BulkNavigationPagePO);
+    await bulkNavigationPagePO.clickNavigateAwait();
+
+    await expect(await appPO.getViewTabCount()).toEqual(11);
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply projects/scion/e2e-testing/src/workbench/bulk-navigation.e2e-spec.ts:23]: Done

File: apps/workbench-client-testing-app/src/app/bulk-navigation-page/bulk-navigation-page.component.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import {Component} from '@angular/core';
+import {WorkbenchNavigationExtras, WorkbenchRouter} from '@scion/workbench-client';
+
+const VIEWS_COUNT = 10;
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply apps/workbench-client-testing-app/src/app/bulk-navigation-page/bulk-navigation-page.component.ts:14]: Done

File: apps/workbench-client-testing-app/src/app/bulk-navigation-page/bulk-navigation-page.component.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    }
+  }
+
+  public async onNavigateAwait(): Promise<void> {
+    for (let i = 0; i < VIEWS_COUNT; i++) {
+      await this.navigateToViewPage();
+    }
+  }
+
+  private navigateToViewPage(): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply apps/workbench-client-testing-app/src/app/bulk-navigation-page/bulk-navigation-page.component.ts:40]: Done

File: projects/scion/e2e-testing/src/workbench-client/page-object/bulk-navigation-page.po.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  public async clickNavigateNoAwait(): Promise<void> {
+    await assertElementVisible(this._locator);
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply projects/scion/e2e-testing/src/workbench-client/page-object/bulk-navigation-page.po.ts]: No, we can it remove from all tests

**Review by danielwiehl — CHANGES_REQUESTED** [review]
Almost perfect. Please see me review commit.

**Review by danielwiehl — APPROVED** [review]

**Pull request merged**